assert/assertions.go

```diff
@@ -1174,6 +1174,46 @@ func formatListDiff(listA, listB interface{}, extraA, extraB []interface{}) stri
 	return msg.String()
 }
 
+// NoElementsMatch asserts that the specified listA(array, slice...)
+// doesn't have any elements in common with the listB(array, slice...).
+//
+// assert.NoElementsMatch(t, [1, 2, 3], [4, 5, 6]) -> true
+//
+// assert.NoElementsMatch(t, [1, 2, 3], [1, 4, 5]) -> false
+func NoElementsMatch(t TestingT, listA, listB interface{}, msgAndArgs ...interface{}) (ok bool) {
+	if h, ok := t.(tHelper); ok {
+		h.Helper()
+	}
+	if isEmpty(listA) || isEmpty(listB) {
+		return true
+	}
+
+	if !isList(t, listA, msgAndArgs...) || !isList(t, listB, msgAndArgs...) {
+		return false
```

Comment:
We mustn't return false, always return a call to Fail.

Reply:
I was referring to its sibling function `ElementsMatch` that also return `true` or `false`   
https://github.com/stretchr/testify/blob/master/assert/assertions.go#L1097

Reply:
adjusted to return a failure call to `Fail`